modules/services/nix-gc/default.nix

```diff
@@ -40,6 +40,12 @@ in
         description = "The time interval at which the garbage collector will run.";
       };
 
+      intervals = mkOption {
+        type = types.listOf types.attrs;
+        default = [cfg.interval];
```

Comment:
I don't remember exactly what the behaviour for this is but I seem to recall that defaults is generally only used for static values and a dynamic value that dynamic default values are defined using `mkDefault` in the config so the priority of the value is lowered and gets replaced when the user specifies a value as expected.

```nix
{
  options = {
    # ...
    intervals = mkOption {
      type = types.listOf types.attrs;
      default = [];
    };
  };
  config = mkIf cfg.automatic {
    # ...
    intervals = mkDefault [cfg.interval];
  };
}
```

Reply:
i just noticed that i haven't pushed a local change to the PR branch. seems like i squashed too so i opted for force-pushing. the behavior i had locally is equivalent to what you've described. PTAL